### PR TITLE
Update docs with DAFpay info

### PR DESCRIPTION
**Pull request by salo1serfati**
Ticket here: https://www.notion.so/givechariot/API-Doc-Changes-b97a3d062eca406ca6c5f63db8b059ea?pvs=4

1. Update `Get Nonprofit` endpoint: 1. If a suborganization is found, it should be returned in this call. @Matthew Magaldi and Salo to flesh out where this will be. 2. If a Nonprofit is not **registered** with Chariot, this should return 404 Not Found.
2. Update `Create Nonprofit` API endpoint. 1. Make the email field optional based on a workflow config. 2. If the Nonprofit they are trying to create is not a 501c3 or pub_78 verified, then throw a **422 Unprocessable Content** with a reason 1.  `The EIN is not a 501c3` 2. `The EIN is not Publication 78 verified` 3. @Matthew Magaldi and Salo to flesh out what this does behind the scenes
3. Update `Create Connect` endpoint 1. Add an optional parameter `suborganization` that allows you to create a Connect for a sub-organization
4. Update `Create Grant` endpoint. 1. Add parameter `applicationFeeAmount` which is the amount in cents that the FP wants to charge for this grant. This fee in addition to Chariot’s fee cannot exceed 5% of the grants amount.
5. Update `Grant` object to return the new set of grant statues
6. Update `Connect` object 1. Add boolean `isDafpayNetwork` parameter, specifying that grants from this connect will go through the DAFpay network. 2. To better explain the userFriendlyId
7. Add Manual grants endpoint to the api docs.

### CODE REVIEW

File: specs/v1.yaml [hunk 1]
```diff
@@ -66,8 +67,9 @@ paths:
         Create a nonprofit organization account.
         This is useful for integration partners to use after a nonprofit consents to use the Chariot payment option on their donation forms.
         If a nonprofit does not already exist for the EIN, this will return a 201 Created status.
-        If a nonprofit already exists for the given EIN on the system, this will return a 409 Conflict status.
-        When integrating this API, it's useful to either handle the 409 error or make a call to the GET v1/nonprofit/{ein} API first to check if it exists prior to creating it.
+        If a nonprofit already exists for the given EIN on the system, this will return a 200 Created status.
+        If the nonprofit does not pass our compliance checks, a 422 Unprocessable Content is returned with a reason.
+        When integrating this API, it's useful to make a call to the GET v1/nonprofit/{ein} API first to check if it exists prior to creating it.
```
Comment (magaldima) [inline specs/v1.yaml:72]: we can remove this line since this should be renamed to "Get or Create"

File: specs/v1.yaml [hunk 2]
```diff
@@ -364,6 +377,89 @@ paths:
           $ref: "#/components/responses/NotFoundError"
         '500':
           $ref: "#/components/responses/InternalServerError"
+  /v1/grants/unintegrated:
```
Comment (magaldima) [inline specs/v1.yaml:380]: i personally would like unintegrated grants to live as a separate resource in our APIs: e.g. `unintegrated-grants` so this becomes: `v1/unintegrated-grants`

File: specs/v1.yaml [hunk 3]
```diff
@@ -364,6 +377,89 @@ paths:
...
+          $ref: "#/components/responses/ListUnintegratedGrantsResponse"
+        '400':
+          $ref: "#/components/responses/BadRequestError"
+        '401':
+          $ref: "#/components/responses/AuthenticationError"
+        '403':
+          $ref: "#/components/responses/ForbiddenError"
+        '500':
+          $ref: "#/components/responses/InternalServerError"
+  /v1/grants/unintegrated/{id}:
```
Comment (magaldima) [inline specs/v1.yaml:419]: working off comment above i think this should be: `/v1/unintegrated-grants/{id}`

File: specs/v1.yaml [hunk 4]
```diff
@@ -472,6 +568,57 @@ components:
...
+          format: date-time
+          description: Time when this object was created. Expressed in ISO 8601 format.
+          example: "2020-07-10 15:00:00.000"
+        updatedAt:
+          type: string
+          readOnly: true
+          format: date-time
+          description: Time when this object was last updated. Expressed in ISO 8601 format.
+          example: "2021-07-11 15:34:00.000"
+    Suborganizations:
```
Comment (magaldima) [inline specs/v1.yaml:598]: the model should be singular, e.g. `Suborganization` and you should make sure to update the reference above

File: specs/v1.yaml [hunk 5]
```diff
@@ -556,6 +703,10 @@ components:
           description: A map of arbitrary string keys and values to store information about the object.
           additionalProperties:
             type: string
+        isDafpayNetwork:
```
Comment (magaldima) [inline specs/v1.yaml:706]: this needs to be moved to the Organization object

File: specs/v1.yaml [hunk 6]
```diff
@@ -556,6 +703,10 @@ components:
...
+          type: boolean
+          description: A flag to indicate if grants made via this Connect will be processed through the DAFpay Network.
```
Comment (magaldima) [inline specs/v1.yaml:708]: description should be updated to be specific to the organization now

File: specs/v1.yaml [hunk 7]
```diff
@@ -821,6 +1059,13 @@ components:
                   The final grant amount in cents that will be processed by Chariot and submitted to the DAF.
                   This amount must be in whole dollar increments (rounded to the nearest hundred) as currently
                   all DAFs only accept whole dollar grants.
+              applicationFeeAmount:
+                type: number
+                description: |-
+                  This parameter specifies the fee your platform plans to take from the grant in cents. This
+                  parameter only applies for grants made through the DAFpay Network. Before
+                  the payment is made available to the nonprofit, Chariot's fee and the applicationFeeAmount
+                  is deducted. The sum of Chariot's fee and the applicationFeeAmount cannot exceed 5% of the grant's amount.
```
Comment (magaldima) [inline specs/v1.yaml:1068]: is this validation done upfront by our server before submitting the grant? if so, what does this API fail with in the case that this happens, e.g. 400 status code? it's useful to clearly write out the error cases

**Review by magaldima — CHANGES_REQUESTED** [review]
@salo1serfati left a few comments. we also may need to revisit the grant statuses enum too

**Comment by salo1serfati** [comment]
> This parameter specifies the fee your platform

Do you think it's worthwhile to expose isDAFpayNetwork on the API? It probably makes sense to abstract that concept away from the FundraisingPlatform. For them, every payment is made via the DAFpayNetwork.

File: specs/v1.yaml [hunk 8]
```diff
@@ -642,7 +788,7 @@ components:
         userFriendlyId:
           type: string
           readOnly: true
-          description: The user friendly identifier for the Grant
+          description: Often refered to as the "Chariot ID", this is the ID that will be included in the payment from the DAF provider.
```
Comment (magaldima) [inline specs/v1.yaml:791]: refered -> referred . also might need to escape the quotes around Chariot ID

File: specs/v1.yaml [hunk 9]
```diff
@@ -710,6 +870,21 @@ components:
...
+    ContributionFeeDetail:
+      type: object
+      required:
+        - name
+        - amount
+      properties:
+        name:
+          type: string
+          description: The name of the party charging the fee
+          example: "Chariot Fee"
```
Comment (magaldima) [inline specs/v1.yaml:882]: this should just be "Chariot'

File: specs/v1.yaml (hunk 8 above)
Comment (magaldima) [reply specs/v1.yaml:791]: also, are we adding `trackingId` to the grant API?

**Review by magaldima — CHANGES_REQUESTED** [review]
left a few minor comments

**Comment by magaldima** [comment]
also we need to merge conflicts

**Review by magaldima — DISMISSED** [review]

**Review by magaldima — APPROVED** [review]

**Comment by magaldima** [comment]
almost there, a few more linter errors:
```
/github/workspace/specs/v1.yaml
 824:20  error  oas3-valid-schema-example  "example" property type must be string                components.schemas.UnintegratedGrant.properties.userFriendlyId.example
 876:17  error  no-$ref-siblings           $ref must not be placed next to any other properties  components.schemas.UnintegratedGrant.properties.address.type
 878:24  error  no-$ref-siblings           $ref must not be placed next to any other properties  components.schemas.UnintegratedGrant.properties.address.description
```

**Review by magaldima — APPROVED** [review]

**Pull request merged**